File: mobile/ios/packages/town-crier-presentation/Sources/Formatters/RadiusFormatter.swift
import Foundation

/// Formats a radius in metres for user-facing display.
///
/// - Values under 1000 m show as whole metres (e.g. "500m").
/// - Values of 1000 m or above show as kilometres, with one decimal
///   place for fractional values (e.g. "1.5km") and no decimal
///   for whole values (e.g. "2km").
///
/// No space before the unit: this string renders in `TCTypography.mono`
/// (SF Mono), where the space glyph is digit-width and reads as a visible
/// double space (GH#912 Phase 1).
public func formatRadius(_ metres: Double) -> String {
  if metres >= 1000 {
    let km = metres / 1000
    if km.truncatingRemainder(dividingBy: 1) == 0 {
      return "\(Int(km))km"
    }
    return String(format: "%.1fkm", km)
  }
  return "\(Int(metres))m"
}

File: mobile/android/presentation/src/main/kotlin/uk/towncrierapp/presentation/watchzones/RadiusFormatter.kt
package uk.towncrierapp.presentation.watchzones

import java.util.Locale

/**
 * Formats a watch-zone radius (metres) for display. Byte-exact port of iOS
 * `RadiusFormatter`'s branching — metric only, no miles:
 *
 * - `< 1000 m` shows as whole metres ("500 m").
 * - `>= 1000 m` shows as kilometres: one decimal place when fractional
 *   ("1.5 km"), no decimal when whole ("2 km").
 *
 * [Locale.UK] is pinned explicitly (rather than the device default, as iOS's
 * un-parameterised `String(format:)` does) purely for deterministic JVM unit
 * tests — this is a UK-only, metric-only app, so the decimal-separator
 * difference a non-UK JVM default locale could otherwise introduce has no
 * user-visible effect.
 */
public object RadiusFormatter {
    public fun format(metres: Double): String {
        if (metres >= 1_000.0) {
            val km = metres / 1_000.0
            return if (km % 1.0 == 0.0) {
                "${km.toInt()} km"
            } else {
                String.format(Locale.UK, "%.1f km", km)
            }
        }
        return "${metres.toInt()} m"
    }
}
